framework/src/systems/LinearSystem.C

```diff
@@ -0,0 +1,250 @@
+    {
+      MatSetOption(petsc_matrix->mat(),
+                   MAT_KEEP_NONZERO_PATTERN, // This is changed in 3.1
+                   PETSC_TRUE);
+      if (!_fe_problem.errorOnJacobianNonzeroReallocation())
+        MatSetOption(petsc_matrix->mat(), MAT_NEW_NONZERO_ALLOCATION_ERR, PETSC_FALSE);
+    }
+  }
+
+  // linear contributions from the domain
+  PARALLEL_TRY
+  {
+    TIME_SECTION("LinearFVKernels_FullSystem", 3 /*, "Computing LinearFVKernels"*/);
+
+    using ElemInfoRange = StoredRange<MooseMesh::const_elem_info_iterator, const ElemInfo *>;
+    ElemInfoRange elem_info_range(_fe_problem.mesh().ownedElemInfoBegin(),
+                                  _fe_problem.mesh().ownedElemInfoEnd());
+
+    using FaceInfoRange = StoredRange<MooseMesh::const_face_info_iterator, const FaceInfo *>;
+    FaceInfoRange face_info_range(_fe_problem.mesh().ownedFaceInfoBegin(),
+                                  _fe_problem.mesh().ownedFaceInfoEnd());
+
+    ComputeLinearFVGreenGaussGradientFaceThread gradient_face_thread(
+        _fe_problem, _fe_problem.linearSysNum(name()));
+    Threads::parallel_reduce(face_info_range, gradient_face_thread);
+
+    ComputeLinearFVGreenGaussGradientVolumeThread gradient_volume_thread(
+        _fe_problem, _fe_problem.linearSysNum(name()));
+    Threads::parallel_reduce(elem_info_range, gradient_volume_thread);
+
+    ComputeLinearFVElementalThread elem_thread(_fe_problem,
+                                               _fe_problem.linearSysNum(name()),
+                                               Moose::FV::LinearFVComputationMode::FullSystem,
+                                               vector_tags);
+    Threads::parallel_reduce(elem_info_range, elem_thread);
+
+    ComputeLinearFVFaceThread face_thread(_fe_problem,
+                                          _fe_problem.linearSysNum(name()),
+                                          Moose::FV::LinearFVComputationMode::FullSystem,
+                                          vector_tags);
+    Threads::parallel_reduce(face_info_range, face_thread);
+  }
+  PARALLEL_CATCH;
+
+  closeTaggedMatrices(matrix_tags);
+
+  _linear_implicit_system.matrix->close();
+  _linear_implicit_system.rhs->close();
+
+  // Accumulate the occurrence of solution invalid warnings for the current iteration cumulative
+  // counters
+  _app.solutionInvalidity().solutionInvalidAccumulation();
+}
+
+NumericVector<Number> &
+LinearSystem::getRightHandSideTimeVector()
+{
+  return *_rhs_time;
+}
+
+NumericVector<Number> &
+LinearSystem::getRightHandSideNonTimeVector()
+{
+  return *_rhs_non_time;
+}
+
+void
+LinearSystem::augmentSparsity(SparsityPattern::Graph & /*sparsity*/,
+                              std::vector<dof_id_type> & /*n_nz*/,
+                              std::vector<dof_id_type> & /*n_oz*/)
+{
+  mooseError("LinearSystem does not support AugmentSparsity!");
+}
+
+void
+LinearSystem::checkKernelCoverage(const std::set<SubdomainID> & /*mesh_subdomains*/) const
+{
```

Comment:
```suggestion
{
// FIXME
```
so it doesnt get missed